src/Storage/OWSOrphanedDataCleaner.m

```diff
@@ -9,82 +11,209 @@
 
 @implementation OWSOrphanedDataCleaner
 
-- (void)removeOrphanedData
++ (void)auditAsync
 {
-    // Remove interactions whose threads have been deleted
-    for (NSString *interactionId in [self orphanedInteractionIds]) {
-        DDLogWarn(@"Removing orphaned interaction with id: %@", interactionId);
-        TSInteraction *interaction = [TSInteraction fetchObjectWithUniqueID:interactionId];
-        [interaction remove];
-    }
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        [OWSOrphanedDataCleaner auditAndCleanup:NO];
+    });
+}
+
++ (void)auditAndCleanupAsync
+{
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        [OWSOrphanedDataCleaner auditAndCleanup:YES];
+    });
+}
 
-    // Remove any lingering attachments
-    for (NSString *path in [self orphanedFilePaths]) {
-        DDLogWarn(@"Removing orphaned file attachment at path: %@", path);
+// This method finds and optionally cleans up:
+//
+// * Orphan messages (with no thread).
+// * Orphan attachments (with no message).
+// * Orphan attachment files (with no attachment).
+// * Missing attachment files (cannot be cleaned up).
```

Comment:
> // * Missing attachment files (cannot be cleaned up).

Do you mean (e.g.) TSAttachmentStream's with no corresponding on disk attachment?

by "(cannot be cleaned up)" I assume you mean we can detect it, but we cannot delete the AttachmentStream? Can you expound on the rationale in the comment?